src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -272,7 +282,12 @@ public static DisposeMethodKind GetDisposeMethodKind(this IMethodSymbol method,
             {
                 if (IsDisposeImplementation(method, iDisposable) ||
                     (Equals(method.ContainingType, iDisposable) &&
-                     method.HasDisposeMethodSignature()))
+                     method.HasDisposeMethodSignature())
+#if NET_ANALYZERS
+                    || (method.ContainingType.IsRefLikeType &&
```

Comment:
I would move the ref struct check inside of the `HasDisposeSignatureByConvention` helper.

Reply:
I'd keep it as is:
1. It would not be immediately clear from the current name, that it also performs type check on the parent type.
2. In both callsites by the time this method is called, it is already known, that type is ref-like, and in one case an extra logic has been computed (scanned for method by name).

Reply:
Makes sense.